service/s3/internal/customizations/response_error_middleware.go

```diff
@@ -0,0 +1,80 @@
+package customizations
+
+import (
+	"context"
+
+	awshttp "github.com/aws/aws-sdk-go-v2/aws/transport/http"
+	"github.com/awslabs/smithy-go/middleware"
+	smithyhttp "github.com/awslabs/smithy-go/transport/http"
+
+	s3http "github.com/aws/aws-sdk-go-v2/service/s3/transport/http"
+)
+
+// AddResponseErrorWrapper adds response error wrapper middleware
+func AddResponseErrorWrapper(stack *middleware.Stack) {
+	// add error wrapper middleware before operation deserializers so that it can wrap the error response
+	// returned by operation deserializers
+	stack.Deserialize.Insert(&errorWrapperMiddleware{}, "OperationDeserializer", middleware.Before)
+}
+
+type errorWrapperMiddleware struct {
+}
+
+// ID returns the middleware identifier
+func (m *errorWrapperMiddleware) ID() string {
+	return "AWSResponseErrorWrapperMiddleware"
+}
```

Comment:
Should this middleware have a different ID than the generic AWSResponseErrorWrapperMiddleware?